src/scripts/main.js

```diff
@@ -1,3 +1,65 @@
 'use strict';
 
 // write code here
+const table = document.querySelector('table');
+const headerOfTable = table.querySelector('thead');
+
+if (headerOfTable) {
+  headerOfTable.addEventListener('click', (e) => {
+    const titles = [...headerOfTable.querySelectorAll('th')];
+    const clickedHeader = e.target.closest('th');
+
+    if (!clickedHeader) {
+      return;
+    }
+
+    const tableData = [...table.querySelectorAll('tbody tr')].map((row) => {
+      return [...row.querySelectorAll('td')].map((cell) => {
+        return cell.textContent.trim();
+      });
+    });
+
+    const index = titles.indexOf(e.target);
+
+    if (index === -1) {
+      return;
+    }
+
+    const indexOfSalaryCell = 3;
```

Comment:
The variable `indexOfSalaryCell` is hardcoded to 3, which assumes that the salary column is always the fourth column. This might not be flexible if the table structure changes. Consider dynamically determining this index or using a data attribute to identify the salary column.